lib/routes/rides/routes.js

```diff
@@ -0,0 +1,118 @@
+import get from 'lodash/get';
+import { notFound, badImplementation, badRequest } from '@utils/responseInterceptors';
+import Joi from 'joi';
+import { bookARide, getPendingRides, assignDriverToRide } from '@daos/ridesDao';
+import { customerIdSchema } from '@utils/validationUtils';
+
+export default [
+  {
+    method: 'POST',
+    path: '/book/{customerId}',
+    options: {
+      description: 'Book a ride for a user',
+      notes: 'Book a ride for a user API',
+      tags: ['api', 'rides'],
+      validate: {
+        params: Joi.object({
+          customerId: customerIdSchema 
+        }),
+        payload: Joi.object({
+          pickUpLocation: Joi.string().trim().required().error(new Error('Pickup location is required.')),
+          dropLocation: Joi.string().trim().required().error(new Error('Drop location is required.'))
```

Comment:
Let's move repetitive schema into `validationUtils.js`
```js
const trimmedStringSchema = Joi.string().trim().required()
```
And while implementing
```js
trimmedStringSchema.error(new Error('Drop location is required.'))
```